src/main/java/fi/helsinki/cs/tmc/core/communication/oauth2/PasswordFlow.java

```diff
@@ -0,0 +1,41 @@
+package fi.helsinki.cs.tmc.core.communication.oauth2;
+
+import fi.helsinki.cs.tmc.core.configuration.TmcSettings;
+import fi.helsinki.cs.tmc.core.holders.TmcSettingsHolder;
+
+import org.apache.oltu.oauth2.client.OAuthClient;
+import org.apache.oltu.oauth2.client.URLConnectionClient;
+import org.apache.oltu.oauth2.client.request.OAuthClientRequest;
+import org.apache.oltu.oauth2.client.response.OAuthJSONAccessTokenResponse;
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+import org.apache.oltu.oauth2.common.message.types.GrantType;
+
+/**
+ * A password flow for oauth.
+ *
+ * <p>Password flow fetches the token from server by using username and password</p>
+ */
+public class PasswordFlow implements OauthFlow {
+
+    public PasswordFlow() {
+    }
+
+    @Override
+    public String getToken() throws OAuthSystemException, OAuthProblemException {
+        TmcSettings settings = TmcSettingsHolder.get();
```

Comment:
If that's the case, maybe we should replace the static oauth with the `getInstance` pattern